tests/robot-tests/tests/admin_and_public/bau/data_catalogue.robot

```diff
@@ -212,6 +212,22 @@ Remove release filter
     user checks page does not contain button    ${PUPIL_ABSENCE_RELEASE_NAME}
     user checks selected option label    id:filters-form-release    All releases
 
+Filter by geographic level
+    user wait for option to be available and select it    css:select[id="filters-form-geographic-level"]
+    ...    Local Authority District
+
+    user checks page contains button    Local Authority District
+    user checks testid element contains    total-results    1 data set
```

Comment:
Just wondering how resilient this is going to be when run in parallel with other test suites where other files may be uploaded containing the same geographic level data. I think there's no filtering by theme or publication applied by this stage so it's operating on all published data data set files. Isn't there a possibility of finding more than one result, especially when run against the Dev environment?

Wouldn't it be better to be filtered by the suite's publication by this stage and check the count of files filtered by geographic level against what we expect, out of the four data set files which were uploaded at the beginning?

We could also check the results are those expected by title or at least have the expected level value displayed in  'Geographic level' in the 'Show more details' section.

Reply:
>  I think there's no filtering by theme or publication applied by this stage

There is still filtering for the pupils and schools theme and pupil absence publication at this point. The previous test case (Remove release filter) just removes the release filter, but leaves the theme and publication filter. So I think this is fine, as the publication is seed data and we'd not expect anyone to add data sets to it.

I will check the the Geographic level of the data set that is returned.

Reply:
> There is still filtering for the pupils and schools theme and pupil absence publication at this point. The previous test case (Remove release filter) just removes the release filter, but leaves the theme and publication filter.

Cool, I missed that 🙂 . Change looks good.